tests/functional/event_handler/test_api_gateway.py

```diff
@@ -0,0 +1,340 @@
+    app = ApiGatewayResolver()
+    mock_event = {"path": "/my/request", "httpMethod": "GET", "headers": {"Accept-Encoding": "deflate, gzip"}}
+    expected_value = '{"test": "value"}'
+
+    @app.get("/my/request", compress=True)
+    def with_compression() -> Tuple[int, str, str]:
+        return 200, APPLICATION_JSON, expected_value
+
+    def handler(event, context):
+        return app.resolve(event, context)
+
+    # WHEN calling the event handler
+    result = handler(mock_event, None)
+
+    # THEN then gzip the response and base64 encode as a string
+    assert result["isBase64Encoded"] is True
+    body = result["body"]
+    assert isinstance(body, str)
+    decompress = zlib.decompress(base64.b64decode(body), wbits=zlib.MAX_WBITS | 16).decode("UTF-8")
+    assert decompress == expected_value
+    headers = result["headers"]
+    assert headers["Content-Encoding"] == "gzip"
+
+
+def test_base64_encode():
+    # GIVEN a function that returns bytes
+    app = ApiGatewayResolver()
+    mock_event = {"path": "/my/path", "httpMethod": "GET", "headers": {"Accept-Encoding": "deflate, gzip"}}
+
+    @app.get("/my/path", compress=True)
+    def read_image() -> Tuple[int, str, bytes]:
+        return 200, "image/png", read_media("idempotent_sequence_exception.png")
+
+    # WHEN calling the event handler
+    result = app(mock_event, None)
+
+    # THEN return the body and a base64 encoded string
+    assert result["isBase64Encoded"] is True
+    body = result["body"]
+    assert isinstance(body, str)
+    headers = result["headers"]
+    assert headers["Content-Encoding"] == "gzip"
+
+
+def test_compress_no_accept_encoding():
+    # GIVEN a function with compress=True
+    # AND the request has no "Accept-Encoding" set to include gzip
+    app = ApiGatewayResolver()
+    expected_value = "Foo"
+
+    @app.get("/my/path", compress=True)
+    def return_text() -> Tuple[int, str, str]:
+        return 200, "text/plain", expected_value
+
+    # WHEN calling the event handler
+    result = app({"path": "/my/path", "httpMethod": "GET", "headers": {}}, None)
+
+    # THEN don't perform any gzip compression
+    assert result["isBase64Encoded"] is False
+    assert result["body"] == expected_value
+
+
+def test_cache_control_200():
+    # GIVEN a function with cache_control set
+    app = ApiGatewayResolver()
+
+    @app.get("/success", cache_control="max-age=600")
+    def with_cache_control() -> Tuple[int, str, str]:
+        return 200, TEXT_HTML, "has 200 response"
+
+    def handler(event, context):
+        return app.resolve(event, context)
+
+    # WHEN calling the event handler
+    # AND the function returns a 200 status code
+    result = handler({"path": "/success", "httpMethod": "GET"}, None)
+
+    # THEN return the set Cache-Control
+    headers = result["headers"]
+    assert headers["Content-Type"] == TEXT_HTML
+    assert headers["Cache-Control"] == "max-age=600"
+
+
+def test_cache_control_non_200():
```

Comment:
Could you also test for CORS being set on non-200, too?

Otherwise, the browser will get a Internal Server Error due to missing some CORS Headers, despite receiving a 400/500 from Lambda in this case.

Reply:
@heitorlessa CORS headers should always be set, but i will at a test to enforce that.

Reply:
Perhaps it's more explicit to have a method for CORS where you could have optional kwargs to override origin (ala FastAPI), etc.  Reasoning is that this would be across all method routes and one can override with headers in Response when necessary (avoid having to add CORS on every method route)

Reply:
@heitorlessa  ok, maybe we can flag the CORS part as `Alpha`, and get some feedback on it.

Reply:
Should I just add a test case to very this is set for non-200?

Reply:
I would assume for `ValueError(f"No route found for '{method}.{path}'")` case we should rather return a 404 and include the CORS if needed

Reply:
If CORS was configured we should return on error too, otherwise the browser will interpret as 500 due to lack of headers despite being a 404

Reply:
Ok i will add that as well as 404 handling for unmapped routes. 

And a 500 error for unhandled exceptions?